www/js/lib/zimfile.js

```diff
@@ -110,6 +110,42 @@ define(['xzdec_wrapper', 'util', 'utf8', 'q', 'zimDirEntry'], function(xz, util,
         }
     };
 
+    /**
+     * Reads the whole MIME type list and returns it as a populated Map
+     * The mimeTypeMap is extracted once after the user has picked the ZIM file
+     * and is stored as ZIMFile.mimetypes.
+     * 
+     * @returns {Promise} A promise for the MIME Type list as a Map
+     */
+    ZIMFile.prototype._mimeTypeMap = function() {
+        var typeMap = new Map;
+        return this._readSlice(this.mimeListPos, 256).then(function(data) {
+            // DEV: We have read 256 bytes: increase this if you encounter longer MIME type lists
+            // also change "while (pos < 255)" below
```

Comment:
The largest MIME type list I have found, in a recent ZIM, is 141 bytes. Is 256 bytes enough of a safety margin, or should I increase to 512?

Reply:
In fileimpl.cpp (of libzim), they seem to know the exact size of what they need to read : getUrlPtrPos() - getMimeListPos()
We can probably do the same, which would be safer than to assume a maximum length.

Reply:
I thought you might say that! I didn't know how to calculate it. Thanks for pointing me in the right direction in libzim. I always get lost whenever I look there (I don't know C++).